core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -149,6 +151,26 @@ public void start(final Listener<RespT> observer, Metadata headers) {
     checkNotNull(observer, "observer");
     checkNotNull(headers, "headers");
 
+    if (deadlineCancellationExecutor == null) {
```

Comment:
I'm not excited about having a very different code path during the race vs after (with this case being after). It makes it all the more difficult to find bugs during the race.